eclair-core/src/main/scala/fr/acinq/eclair/transactions/Transactions.scala

```diff
@@ -206,39 +207,97 @@ object Transactions {
+        htlcA.cltvExpiry <= htlcB.cltvExpiry
+      case _ => LexicographicalOrdering.isLessThan(a.output, b.output)
+    }
+  }
+
+  def makeCommitTxOutputs(localIsFunder: Boolean,
+                          localDustLimit: Satoshi,
+                          localRevocationPubkey: PublicKey,
+                          toLocalDelay: CltvExpiryDelta,
+                          localDelayedPaymentPubkey: PublicKey,
+                          remotePaymentPubkey: PublicKey,
+                          localHtlcPubkey: PublicKey,
+                          remoteHtlcPubkey: PublicKey,
+                          spec: CommitmentSpec): Seq[CommitmentOutputLink]  = {
     val commitFee = commitTxFee(localDustLimit, spec)
 
     val (toLocalAmount: Satoshi, toRemoteAmount: Satoshi) = if (localIsFunder) {
       (spec.toLocal.truncateToSatoshi - commitFee, spec.toRemote.truncateToSatoshi)
     } else {
       (spec.toLocal.truncateToSatoshi, spec.toRemote.truncateToSatoshi - commitFee)
     } // NB: we don't care if values are < 0, they will be trimmed if they are < dust limit anyway
-
-    val toLocalDelayedOutput_opt = if (toLocalAmount >= localDustLimit) Some(TxOut(toLocalAmount, pay2wsh(toLocalDelayed(localRevocationPubkey, toLocalDelay, localDelayedPaymentPubkey)))) else None
-    val toRemoteOutput_opt = if (toRemoteAmount >= localDustLimit) Some(TxOut(toRemoteAmount, pay2wpkh(remotePaymentPubkey))) else None
-
-    val htlcOfferedOutputs = trimOfferedHtlcs(localDustLimit, spec)
-      .map(htlc => TxOut(htlc.add.amountMsat.truncateToSatoshi, pay2wsh(htlcOffered(localHtlcPubkey, remoteHtlcPubkey, localRevocationPubkey, ripemd160(htlc.add.paymentHash.bytes)))))
-    val htlcReceivedOutputs = trimReceivedHtlcs(localDustLimit, spec)
-      .map(htlc => TxOut(htlc.add.amountMsat.truncateToSatoshi, pay2wsh(htlcReceived(localHtlcPubkey, remoteHtlcPubkey, localRevocationPubkey, ripemd160(htlc.add.paymentHash.bytes), htlc.add.cltvExpiry))))
-
+    val outputs = collection.mutable.ArrayBuffer.empty[CommitmentOutputLink]
+
+    if (toLocalAmount >= localDustLimit) outputs.append(
+      CommitmentOutputLink(
+        TxOut(toLocalAmount, pay2wsh(toLocalDelayed(localRevocationPubkey, toLocalDelay, localDelayedPaymentPubkey))),
+        toLocalDelayed(localRevocationPubkey, toLocalDelay, localDelayedPaymentPubkey),
+        CommitmentOutput.ToLocal))
+
+    if (toRemoteAmount >= localDustLimit) outputs.append(
+      CommitmentOutputLink(
+        TxOut(toRemoteAmount, pay2wpkh(remotePaymentPubkey)),
+        pay2pkh(remotePaymentPubkey),
+        CommitmentOutput.ToRemote))
+
+    trimOfferedHtlcs(localDustLimit, spec).foreach(htlc =>
+      outputs.append(CommitmentOutputLink(
+        TxOut(htlc.add.amountMsat.truncateToSatoshi, pay2wsh(htlcOffered(localHtlcPubkey, remoteHtlcPubkey, localRevocationPubkey, ripemd160(htlc.add.paymentHash.bytes)))),
```

Comment:
Maybe worth removing the duplicated `htlcOffered()`?

```scala
val htlcScript = htlcOffered(localHtlcPubkey, remoteHtlcPubkey, localRevocationPubkey, ripemd160(htlc.add.paymentHash.bytes)))
TxOut(htlc.add.amountMsat.truncateToSatoshi, pay2wsh(htlcScript), htlcScript, htlc)
```

And same below for htlc received?